apps/api/src/app/workflows-v2/usecases/list-workflows/list-workflows.command.ts

```diff
@@ -0,0 +1,10 @@
+import { EnvironmentWithUserObjectCommand } from '@novu/application-generic';
+import { IsDefined } from 'class-validator';
+
+export class ListWorkflowsCommand extends EnvironmentWithUserObjectCommand {
+  searchQuery?: string;
+  @IsDefined()
+  offset?: number;
+  @IsDefined()
+  limit?: number;
```

Comment:
```suggestion
  @IsOptional()
  @IsString()
  searchQuery?: string;

  @IsDefined()
  @IsNumber()
  offset: number;

  @IsDefined()
  @IsNumber()
  limit: number;
```

Reply:
fix spacing

Reply:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Could we abstract this class into a  utility`ListResourceCommand` that other resources can extend to ensure our API pagination is implemented consistently across all resources?‏

Reply:
Love the idea, we should do something like we have on the API level [PaginationRequestDto](https://github.com/novuhq/novu/blob/nv-4282-post-v2-workflows/apps/api/src/app/shared/dtos/pagination-request.ts#L8)

Reply:
will do